apps/cli/src/prebuild.ts

```diff
@@ -43,7 +51,30 @@ type RemixRoutes = {
   }>;
 };
 
+export const downloadAsset = async (url: string, name: string) => {
+  const assetPath = join("public", ASSETS_BASE, name);
+  try {
+    await access(assetPath);
+  } catch {
+    const streamPipeline = promisify(pipeline);
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`unexpected response ${response.statusText}`);
+      }
+
+      await streamPipeline(
+        response.body as NodeJS.ReadableStream,
+        createWriteStream(assetPath)
+      );
+    } catch (error) {
+      console.error(`Error in downloading file ${name} \n ${error}`);
```

Comment:
Best would be to make cleanup, as here file can be created but be broken

Reply:
Also good thing to reduce various errors caused by cli termination etc is to download files initially to  newly created tmp dir (there is a command to get system tmp dir location). https://www.npmjs.com/package/tmp
and immediately after download move file to the destination.

What it gives.

Download is a long operation, every termination of process will leave you with partially downloaded file, that's bad because on resync we will not download it again.
Move is superfast.

With above idea if termination occur on download then nothing bad happens. It was download to tmp. And the probability of termination on download is huge, because of time.

Probability of termination on file move is near zero. Because it's near immediate system operation.

Reply:
There is an old lib for this https://www.npmjs.com/package/fs-write-stream-atomic
сс @TrySound can you recommend anything else?

Reply:
https://nodejs.org/dist/latest-v18.x/docs/api/fs.html#fspromisesmkdtempprefix-options